tokio-console/src/config.rs

```diff
@@ -107,20 +112,62 @@ pub struct ViewOptions {
 }
 
 /// Toggles on and off color coding for individual UI elements.
-#[derive(Clap, Debug, Copy, Clone)]
+#[derive(Clap, Debug, Copy, Clone, Deserialize)]
 pub struct ColorToggles {
     /// Disable color-coding for duration units.
-    #[clap(long = "no-duration-colors", parse(from_flag = std::ops::Not::not), group = "colors")]
-    pub(crate) color_durations: bool,
+    #[clap(long = "no-duration-colors", group = "colors")]
+    #[serde(rename = "durations")]
+    color_durations: Option<bool>,
 
     /// Disable color-coding for terminated tasks.
-    #[clap(long = "no-terminated-colors", parse(from_flag = std::ops::Not::not), group = "colors")]
-    pub(crate) color_terminated: bool,
+    #[clap(long = "no-terminated-colors", group = "colors")]
+    #[serde(rename = "terminated")]
+    color_terminated: Option<bool>,
+}
+
+/// A sturct used to parse the toml config file
+#[derive(Debug, Clone, Deserialize)]
+struct ConfigFile {
+    charset: Option<CharsetConfig>,
+    colors: Option<ColorsConfig>,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+struct CharsetConfig {
+    lang: Option<String>,
+    ascii_only: Option<bool>,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+struct ColorsConfig {
+    enabled: Option<bool>,
+    truecolor: Option<bool>,
+    palette: Option<Palette>,
+    enable: Option<ColorToggles>,
 }
 
 // === impl Config ===
 
 impl Config {
+    /// Parse from config files and command line options.
+    pub fn parse() -> color_eyre::Result<Self> {
```

Comment:
one last naming nit, take it or leave it: I'm not the _biggest_ fan of calling this `parse`, as it conflicts with the `Clap::parse` implementation, which is a bit confusing. but, I can't immediately think of a better name for it, so :woman_shrugging: